src/model/clients/HttpClient.js

```diff
@@ -0,0 +1,133 @@
+import axios from 'axios';
+import { AuthorizationTypes } from 'shared';
+
+export const HttpRequestMethods = {
+  GET: 'GET',
+  POST: 'POST',
+  PUT: 'PUT',
+  PATCH: 'PATCH',
+  DELETE: 'DELETE',
+};
+
+export class HttpClient {
+  static get(url, config) {
+    const boundConfig = Object.assign({}, config, {
+      url,
+      method: HttpRequestMethods.GET,
+    });
+    const newClient = new HttpClient(boundConfig);
```

Comment:
Is it really required to instantiate a new HttpClient for each request?

Reply:
these are only for one-shot requests right? Maybe these shoudn't be supported like this. My main concern would be to have too many floating around clients but I'm unsure if this will be a problem for browsers :)

Reply:
The clients are declared inside function bodies, so they'll get marked for garbage collection once the function gets done executing (in this case when the promise resolves or rejects), so memory usage is not a real issue.
Both `ky` and `axios` do the same thing behind de scenes, it's not a big deal.

With the current implementation you could re-use one if you really wanted, but since the configuration of the request is stored as an instance property of the class, you'd have to reset it for every use.

Reusing would look like this:
```javascript
const client = new HttpClient({
  method: HttpRequestMethods.GET,
  url: requestURL,
});
client.setAuthorizationToken(authToken);

const firstResponse = await client.execute();

client.setRequestConfig({
  method: HttpRequestMethods.POST,
  url: requestURL,
  body: data,
});
client.setAuthorizationToken(authToken);

const secondResponse = await client.execute();

```

Please keep in mind that the static methods are factories for quickly executing a request, they are not the only way to use them.

We could have the client as a function, and then pass everything to it. And for the `GiantSwarmClient`, we would just create a new function and return the client function with a few bound values.
For me, there are 2 problems with that approach:
* The developer experience would not be as good as using instance methods for configuration
* The interface would not be the same for other clients (such as a WebSockets client - where we would need to keep a reference of the connection, so we could stop it on demand)

I'm not disregarding the idea of switching to a function, but please take the things above into consideration

Reply:
Cool thanks for the detailed explanation :+1: